### PR TITLE
Bjorncs/controlled deconstruction of old graph [run-systemtest]

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @gjoranv @jonmv

### CODE REVIEW

**Review by jonmv — APPROVED** [review]
Exciting times!

**Review by baldersheim — COMMENTED** [review]
👍

**Review by gjoranv — COMMENTED** [review]
Great!

File: container-core/src/main/java/com/yahoo/container/core/config/HandlersConfigurerDi.java [hunk 1]
```diff
@@ -106,11 +107,15 @@ public void installPlatformBundles(Collection<String> bundlePaths) {
...
-    public void waitForNextComponentGeneration(Injector discInjector, boolean isInitializing) {
-        currentGraph = container.waitForNextComponentGeneration(currentGraph,
-                                                      createFallbackInjector(vespaContainer, discInjector),
-                                                      isInitializing);
+    public Runnable waitForNextComponentGeneration(Injector discInjector, boolean isInitializing) {
+        Container.ComponentGraphResult result = container.waitForNextComponentGeneration(
+                this.currentGraph,
+                createFallbackInjector(vespaContainer, discInjector),
+                isInitializing);
+        this.currentGraph = result.newGraph();
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/container/core/config/HandlersConfigurerDi.java:117]: The cleanup task may not necessarily be executed in chronological order. Only the caller, `ConfiguredApplication`, knows when the previous components can be safely deconstructed. `HandlersConfigurerDi` may store the cleanup tasks if indexed on config generation

**Pull request merged**